iib/web/api_v1.py

```diff
@@ -439,21 +444,38 @@ def regenerate_bundle_batch():
     # This list will be used for the log message below and avoids the need of having to iterate
     # through the list of requests another time
     request_id_strs = []
-    for build_request, request in zip(payload['build_requests'], requests):
-        request_jsons.append(request.to_json())
-        request_id_strs.append(str(request.id))
-
-        error_callback = failed_request_callback.s(request.id)
-        handle_regenerate_bundle_request.apply_async(
-            args=[
-                build_request['from_bundle_image'],
-                build_request.get('organization'),
-                request.id,
-            ],
-            link_error=error_callback,
-            queue=_get_user_queue(),
+    build_and_requests = zip(payload['build_requests'], requests)
+    try:
+        for build_request, request in build_and_requests:
+            error_callback = failed_request_callback.s(request.id)
+            handle_regenerate_bundle_request.apply_async(
+                args=[
+                    build_request['from_bundle_image'],
+                    build_request.get('organization'),
+                    request.id,
+                ],
+                link_error=error_callback,
+                queue=_get_user_queue(),
+            )
+
+            request_jsons.append(request.to_json())
+            request_id_strs.append(str(request.id))
+    except OperationalError as e:
+        failed_ids = []
+        for _, request in build_and_requests:
+            if str(request.id) in request_id_strs:
+                continue
+
+            failed_ids.append(str(request.id))
+            request.add_state('failed', f'Scheduling of request failed.')
+            messaging.send_message_for_state_change(request)
+
+        flask.current_app.logger.exception(
+            f'Scheduling of request with ids {", ".join(failed_ids)} failed.\nException: {e}'
```

Comment:
It's unnecessary to log the exception message in the log message itself since you are doing `logger.exception`, which will log the exception and traceback.
```suggestion
            f'Scheduling of request with IDs {", ".join(failed_ids)} failed'
```